### PR TITLE
Remove branch complexity

### PR DESCRIPTION
**Pull request by nogates**
#### Description

What Have I Done?

 - Fix the broke tests
 - Remove method complexity
 - Test that the `skip` is actually working

Related: https://github.com/envato/static_sync/pull/1

### CODE REVIEW

File: lib/static_sync/processor.rb [hunk 1]
```diff
@@ -16,27 +16,14 @@ def initialize(config, storage = nil)
...
-            begin
-              @storage.create(current_file.headers) unless skip(file)
-            rescue => error
-              log.error("Failed to upload #{file}")
-              raise error
-            end
-          end
-        end
+      log_sync
+      each_local_filter_file do |uploadable_file|
```
Comment (taoza) [inline lib/static_sync/processor.rb]: `each_uploadable_file do |uploadable_file|`?

**Comment by joneslee85** [comment]
:+1:

**Comment by joneslee85** [comment]
If possible, please lock the supported Ruby to current supported versions in gemspec

```
gem.required_ruby_version = '>= 2.2.0'
```

and remove references of Ruby 1.9 in README.md

**Comment by fraserxu** [comment]
Thanks for the fix @nogates :+1:

**Comment by nogates** [comment]
> If possible, please lock the supported Ruby to current supported versions in gemspec
> 
> gem.required_ruby_version = '>= 2.2.0'
> and remove references of Ruby 1.9 in README.md

maybe, we do it later. We should get this done ASAP

File: lib/static_sync/processor.rb [hunk 2]
```diff
@@ -65,6 +52,35 @@ def local_filtered_files
 
     private
 
+    def look_for_conflicts(file)
+      if @storage.has_file?(file.version)
+        handle_conflict(file)
+      else
+        log.info("Uploading #{ file }") if @config.log
```
Comment (taoza) [inline lib/static_sync/processor.rb]: Why "Uploading #{file}" is logged in a method called `look_for_conflicts?`

Comment (nogates) [inline lib/static_sync/processor.rb]: Good point. Although, this is intended to be a quick fix. TBH, I'd almost rewrite the whole file

**Comment by taoza** [comment]
> TBH, I'd almost rewrite the whole file
:+1: 

> this is intended to be a quick fix
Let's get this out so we can deploy :) :+1:

**Pull request merged**